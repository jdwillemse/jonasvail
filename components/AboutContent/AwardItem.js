import Link from 'next/link'

import linkResolver from '../../utils/linkResolver'
import css from './styles.module.scss'

const AwardItem = ({ year, title, project }) => {
  return (
    <li className={css.awardWrap}>
      <div className={css.awardDate}>{year}</div>
      <div className={css.awardTitle}>{title}</div>
      {project && (
        <Link
          href={linkResolver({ ...project, type: 'project' })}
          key={project.uid}
        >
          <span className={css.linkInner}>{project.title}</span>
        </Link>
      )}
    </li>
  )
}

export default AwardItem
