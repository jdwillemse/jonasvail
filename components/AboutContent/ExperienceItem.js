'use server'
import React from 'react'

import css from './styles.module.scss'

const ExperienceItem = ({ dates, jobTitle, employer, description }) => {
  return (
    <div className={css.experienceWrap}>
      {dates && <div className={css.awardDate}>{dates}</div>}
      {jobTitle && (
        <div className={!employer && css.experienceEmployer}>{jobTitle}</div>
      )}
      {employer && <div className={css.experienceEmployer}>{employer}</div>}
      {description && (
        <div className={css.experienceDescription}>{description}</div>
      )}
    </div>
  )
}

export default ExperienceItem
