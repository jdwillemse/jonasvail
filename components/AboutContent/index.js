import React from 'react'
// import PropTypes from 'prop-types'

import css from './styles.module.scss'
import RichText from '../RichText'

const AboutContent = ({ experience, awards }) => {
  return (
    <div className="row">
      <div className="col-sm-12 col-md-6">
        <RichText content={experience} className={css.content} />
      </div>
      <div className="col-sm-12 col-md-6">
        <RichText content={awards} className={css.content} />
      </div>
    </div>
  )
}

AboutContent.propTypes = {
  // children: PropTypes.node.isRequired,
}

export default AboutContent
