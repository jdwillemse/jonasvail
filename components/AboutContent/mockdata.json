{
  "experience": [
    { "type": "paragraph", "text": "WORK EXPERIENCE", "spans": [] },
    {
      "type": "paragraph",
      "text": "December 2017 to Present\nFreelance Brand and Advertising Strategist",
      "spans": [{ "start": 25, "end": 67, "type": "strong" }]
    },
    {
      "type": "paragraph",
      "text": "Clients: adidas Brand Design, Gypsy, Inc. for Twitch, 180 Amsterdam for PlayStation and Lay’s, Seventy Agency Stockholm for Ericsson, Tommy Hilfiger\nPitches won: Twitch EMEA\n\n\nApril 2011 – December 2017\nStrategic Planning Director\n180 Amsterdam\n\nClients: Sony PlayStation (PS4, PS VR, games), Hewlett-Packard, Diageo, Lilly Diabetes UK, ASICS\nPitches won: Ketel One, Sky, Lilly, Replay, Qatar Airways, Al Jazeera, VAT 69, PlayStation\n",
      "spans": [{ "start": 231, "end": 244, "type": "strong" }]
    },
    {
      "type": "paragraph",
      "text": "\nSeptember 2007 – April 2011\nStrategic Planning Director\nWieden+Kennedy, Amsterdam\n\nClients: EA SPORTS FIFA, BASF, Nokia\nPitches won: BASF\n\n\nMarch 2005 – June 2007\nPlanning Director\nSaatchi & Saatchi, London\n\nClients: Dr. Martens, T-Mobile, Sagatiba\n\n\nDecember 1999 - March 2005\nSenior Account Planner \nGoodby, Silverstein & Partners, San Francisco\n\nClients: Hewlett-Packard (HP), Saturn (a division of GM), Frito-Lay, Hotwire.com, Epinions.com\n\n\nJanuary 1998 - December 1999\nAccount Planner\nFoote, Cone & Belding, San Francisco\n\nClients: Sega, MTV, Slates (a Levi-Strauss & Co. brand)\n\n\nSeptember 1995 - December 1997\nMarket Research/Product Development\nLambesis, Inc., Del Mar\n\nClients: Cadillac, Airwalk, The NFL, Rossignol, Mattel, Columbia-Tristar, Calvin Klein",
      "spans": [
        { "start": 57, "end": 82, "type": "strong" },
        { "start": 182, "end": 207, "type": "strong" },
        { "start": 232, "end": 233, "type": "strong" },
        { "start": 303, "end": 348, "type": "strong" },
        { "start": 366, "end": 367, "type": "strong" },
        { "start": 413, "end": 414, "type": "strong" },
        { "start": 492, "end": 528, "type": "strong" },
        { "start": 564, "end": 565, "type": "strong" },
        { "start": 655, "end": 678, "type": "strong" },
        { "start": 744, "end": 745, "type": "strong" }
      ]
    }
  ],
  "awards": [
    { "type": "paragraph", "text": "AWARDS & RECOGNITION", "spans": [] },
    {
      "type": "paragraph",
      "text": "2014/15\nGrand Prix and Gold Euro Effie, and Global Effie Finalist\nPS4 Launch campaign\n\n\n2013\nBronze Cannes Lion\nThe Last of Us\n\n\n2010\nSilver Euro Effie\nFIFA 10 campaign\n\n\n2004\nCampaign Magazine’s “International Advertiser of the Year”\n+hp Brand Campaign\n\n\n2003\nCreativity Magazine’s “Campaign of the Year”\n+hp Brand Campaign\n\n\n2000/2001\nBronze and Silver North America Effies\nSega Dreamcast Launch Campaign\n\n\n1997\nMentioned in the New Yorker article “The Coolhunt”, by Malcolm Gladwell",
      "spans": [
        {
          "start": 66,
          "end": 85,
          "type": "hyperlink",
          "data": {
            "id": "XpBc6RAAAO7Gza0K",
            "type": "project",
            "tags": [],
            "lang": "en-gb",
            "slug": "ps4-launch-this-is-for-the-players",
            "first_publication_date": "2020-04-10T11:47:59+0000",
            "last_publication_date": "2020-10-25T14:25:10+0000",
            "uid": "ps4-launch-this-is-for-the-players",
            "link_type": "Document",
            "isBroken": false
          }
        },
        {
          "start": 112,
          "end": 126,
          "type": "hyperlink",
          "data": {
            "id": "XpBiRxAAAO7FzcVW",
            "type": "project",
            "tags": [],
            "lang": "en-gb",
            "slug": "the-last-of-us-could-you-be-the-last-of-us",
            "first_publication_date": "2020-04-10T12:10:54+0000",
            "last_publication_date": "2020-10-28T10:38:20+0000",
            "uid": "could-you-be-the-last-of-us",
            "link_type": "Document",
            "isBroken": false
          }
        },
        {
          "start": 152,
          "end": 169,
          "type": "hyperlink",
          "data": {
            "id": "XdqieBEAACMAEhJt",
            "type": "project",
            "tags": [],
            "lang": "en-gb",
            "slug": "ea-sports-fifa-10-how-big-can-football-get",
            "first_publication_date": "2019-11-24T15:32:11+0000",
            "last_publication_date": "2020-10-28T10:38:20+0000",
            "uid": "ea-sports-fifa10-how-big-can-football-get",
            "link_type": "Document",
            "isBroken": false
          }
        },
        {
          "start": 235,
          "end": 254,
          "type": "hyperlink",
          "data": {
            "id": "XpBmNhAAACzKzdeG",
            "type": "project",
            "tags": [],
            "lang": "en-gb",
            "slug": "hp-everything-is-possible",
            "first_publication_date": "2020-04-10T12:27:39+0000",
            "last_publication_date": "2020-10-21T17:19:11+0000",
            "uid": "hp-everything-is-possible",
            "link_type": "Document",
            "isBroken": false
          }
        },
        {
          "start": 306,
          "end": 325,
          "type": "hyperlink",
          "data": {
            "id": "XpBmNhAAACzKzdeG",
            "type": "project",
            "tags": [],
            "lang": "en-gb",
            "slug": "hp-everything-is-possible",
            "first_publication_date": "2020-04-10T12:27:39+0000",
            "last_publication_date": "2020-10-21T17:19:11+0000",
            "uid": "hp-everything-is-possible",
            "link_type": "Document",
            "isBroken": false
          }
        },
        {
          "start": 376,
          "end": 406,
          "type": "hyperlink",
          "data": {
            "id": "XdqfxxEAACEAEgZ4",
            "type": "project",
            "tags": [],
            "lang": "en-gb",
            "slug": "sega-dreamcast-launch-its-thinking",
            "first_publication_date": "2019-11-24T15:20:43+0000",
            "last_publication_date": "2020-10-27T20:52:16+0000",
            "uid": "sega-dreamcast-its-thinking",
            "link_type": "Document",
            "isBroken": false
          }
        }
      ]
    }
  ]
}
