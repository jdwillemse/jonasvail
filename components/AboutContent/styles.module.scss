@import '../../styles/vars.scss';

.wrap {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  gap: 2rem;
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.col {
  grid-column: 1 / span 12;
}

.list {
  margin: 0;
  padding: 0;
  list-style: none;
}

.link {
  color: $color-dark;
}

.title {
  font-size: 1.25rem;
  margin-bottom: 2rem;
}

.content {
  margin-bottom: 2rem;
}

.image {
  display: block;
  max-width: 100%;
  height: auto;
  margin-bottom: 1rem;
}

.awardWrap,
.experienceWrap {
  margin-bottom: 2.5rem;
}

.experienceEmployer {
  font-weight: bold;
  margin-bottom: 1rem;
}

.experienceEmployer,
.awardTitle {
  font-weight: bold;
}

.experienceDescription {
  margin-top: 0.5rem;
}

@media (min-width: 576px) {
  .col {
    grid-column: 1 / span 6;
  }
  .col:last-child {
    grid-column: 7 / span 6;
  }
}
