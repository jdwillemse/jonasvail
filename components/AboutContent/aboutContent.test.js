import { render } from '@testing-library/react'

import AboutContent from '.'
import data from './mockdata.json'

describe('AboutContent', () => {
  it('renders when content is passed', () => {
    const { container } = render(<AboutContent {...data} />)
    expect(container).toMatchSnapshot()
  })

  it('renders when content is not passed', () => {
    const { container } = render(<AboutContent />)
    expect(container.querySelector('.col-sm-12').childNodes.length).toBe(0)
  })
})
