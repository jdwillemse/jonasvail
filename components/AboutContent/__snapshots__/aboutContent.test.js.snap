// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutContent renders when content is passed 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-sm-12 col-md-6"
    >
      <div
        class="content wrap"
      >
        <p>
          WORK EXPERIENCE
        </p>
        <p>
          December 2017 to Present
          <br />
          <strong>
            Freelance Brand and Advertising Strategist
          </strong>
        </p>
        <p>
          Clients: adidas Brand Design, Gypsy, Inc. for Twitch, 180 Amsterdam for PlayStation and Lay’s, Seventy Agency Stockholm for Ericsson, Tommy Hilfiger
          <br />
          Pitches won: Twitch EMEA
          <br />
          <br />
          <br />
          April 2011 – December 2017
          <br />
          Strategic Planning Director
          <br />
          <strong>
            180 Amsterdam
          </strong>
          <br />
          <br />
          Clients: Sony PlayStation (PS4, PS VR, games), Hewlett-Packard, Diageo, Lilly Diabetes UK, ASICS
          <br />
          Pitches won: Ketel One, Sky, Lilly, Replay, Qatar Airways, Al Jazeera, VAT 69, PlayStation
          <br />
        </p>
        <p>
          <br />
          September 2007 – April 2011
          <br />
          Strategic Planning Director
          <br />
          <strong>
            Wieden+Kennedy, Amsterdam
          </strong>
          <br />
          <br />
          Clients: EA SPORTS FIFA, BASF, Nokia
          <br />
          Pitches won: BASF
          <br />
          <br />
          <br />
          March 2005 – June 2007
          <br />
          Planning Director
          <br />
          <strong>
            Saatchi & Saatchi, London
          </strong>
          <br />
          <br />
          Clients: Dr. Martens, T
          <strong>
            -
          </strong>
          Mobile, Sagatiba
          <br />
          <br />
          <br />
          December 1999 - March 2005
          <br />
          Senior Account Planner 
          <br />
          <strong>
            Goodby, Silverstein & Partners, San Francisco
          </strong>
          <br />
          <br />
          Clients: Hewlett
          <strong>
            -
          </strong>
          Packard (HP), Saturn (a division of GM), Frito
          <strong>
            -
          </strong>
          Lay, Hotwire.com, Epinions.com
          <br />
          <br />
          <br />
          January 1998 - December 1999
          <br />
          Account Planner
          <br />
          <strong>
            Foote, Cone & Belding, San Francisco
          </strong>
          <br />
          <br />
          Clients: Sega, MTV, Slates (a Levi
          <strong>
            -
          </strong>
          Strauss & Co. brand)
          <br />
          <br />
          <br />
          September 1995 - December 1997
          <br />
          Market Research/Product Development
          <br />
          <strong>
            Lambesis, Inc., Del Mar
          </strong>
          <br />
          <br />
          Clients: Cadillac, Airwalk, The NFL, Rossignol, Mattel, Columbia
          <strong>
            -
          </strong>
          Tristar, Calvin Klein
        </p>
      </div>
    </div>
    <div
      class="col-sm-12 col-md-6"
    >
      <div
        class="content wrap"
      >
        <p>
          AWARDS & RECOGNITION
        </p>
        <p>
          2014/15
          <br />
          Grand Prix and Gold Euro Effie, and Global Effie Finalist
          <br />
          <a
            href="/ps4-launch-this-is-for-the-players"
          >
            PS4 Launch campaign
          </a>
          <br />
          <br />
          <br />
          2013
          <br />
          Bronze Cannes Lion
          <br />
          <a
            href="/could-you-be-the-last-of-us"
          >
            The Last of Us
          </a>
          <br />
          <br />
          <br />
          2010
          <br />
          Silver Euro Effie
          <br />
          <a
            href="/ea-sports-fifa10-how-big-can-football-get"
          >
            FIFA 10 campaign
            <br />
          </a>
          <br />
          <br />
          2004
          <br />
          Campaign Magazine’s “International Advertiser of the Year”
          <br />
          <a
            href="/hp-everything-is-possible"
          >
            +hp Brand Campaign
            <br />
          </a>
          <br />
          <br />
          2003
          <br />
          Creativity Magazine’s “Campaign of the Year”
          <br />
          <a
            href="/hp-everything-is-possible"
          >
            +hp Brand Campaign
            <br />
          </a>
          <br />
          <br />
          2000/2001
          <br />
          Bronze and Silver North America Effies
          <br />
          <a
            href="/sega-dreamcast-its-thinking"
          >
            Sega Dreamcast Launch Campaign
          </a>
          <br />
          <br />
          <br />
          1997
          <br />
          Mentioned in the New Yorker article “The Coolhunt”, by Malcolm Gladwell
        </p>
      </div>
    </div>
  </div>
</div>
`;
